### ISSUE TITLE
RuntimeWarning and NaN Values in Loss

### ISSUE THREAD

**Issue by AllenDiiaz** [jungeun122333/UVI-Net#1]
Hi, I am a master's student from National Taiwan Univ, studying in the Department of Medical Engineering. 
I am impressed with your work and would like to seek some advice on a few issues I am encountering!
Two issues while running the training script:

**1. RuntimeWarning:**
I received the following warning during execution:
```python
/home/allen/.conda/envs/UVI/lib/python3.8/site-packages/numpy/core/_methods.py:233: RuntimeWarning: invalid value encountered in subtract
  x = asanyarray(arr - arrmean)
``` 
**2. NaN Values in loss_all/train:**
Additionally, the loss_all/train,loss_full/train metrics are consistently showing NaN values throughout the training process.

I have tried two different versions of the 4D-Lung-Preprocessed dataset (CVPR and update version), but the same problem persists.

**Environment:**

Hardware:
Server: Supermicro
GPU: NVIDIA A100 80GB PCIe

Software:
Operating System: Windows 11 
Remote Access Tool: MobaXterm 23.2
Python/PyTorch Version: Same as specified in the requirements.txt provided by the repository

Thank you for your patience and your prompt responses!

Best,
Allen

![螢幕擷取畫面 2024-08-03 180137](https://github.com/user-attachments/assets/ae9fd843-6798-4f8f-be01-eb902755924a)

**Comment by jungeun122333** [comment]
Dear Allen,

I apologize for the delayed response.

Have you had a chance to work with the cardiac dataset? I would like to determine whether the error is originating from the dataset itself or the code.

**Comment by Zeyad-Zaky** [comment]
I have the same issue, but it happened when I used the lung dataset, however made the preprocessing stop at img sizes (256,256,256) not (128,128,128)

**Comment by iisfive** [comment]
Hello, has the problem been resolved?

**Comment by tuhlnaa** [comment]
> Hello, has the problem been resolved?

Hi @iisfive,
I forked this repository a while ago and didn't encounter these issues. I made some refactoring and bug fixes in my fork, which might be helpful: https://github.com/tuhlnaa/UVI-Net-Extended

**Comment by iisfive** [comment]
> > Hello, has the problem been resolved?
> 
> Hi [@iisfive](https://github.com/iisfive), I forked this repository a while ago and didn't encounter these issues. I made some refactoring and bug fixes in my fork, which might be helpful: https://github.com/tuhlnaa/UVI-Net-Extended

Wow, thank you so much !!! I'll go try it right now! You're such a nice person!!!

**Comment by iisfive** [comment]
> > Hello, has the problem been resolved?
> 
> Hi [@iisfive](https://github.com/iisfive), I forked this repository a while ago and didn't encounter these issues. I made some refactoring and bug fixes in my fork, which might be helpful: https://github.com/tuhlnaa/UVI-Net-Extended

Thank you very much for your work, but it's frustrating that I encountered the same issue in your version.
